testing/code/test_excinfo.py

```diff
@@ -965,7 +965,7 @@ def raiseos():
             assert upframe is not None
             if upframe.f_code.co_name == "_makepath":
                 # Only raise with expected calls, but not via e.g. inspect for
-                # py38-windows.
+                # py38-windows. (?)
                 raised += 1
                 raise OSError(2, "custom_oserror")
             return orig_path_cwd()
```

Comment:
Not sure what to do here, advise welcome

Reply:
https://github.com/pytest-dev/pytest/pull/6529 does not offer any explanation, however https://github.com/pytest-dev/pytest/pull/6463#issuecomment-578092851 links in and implies that there might've been flakiness in the Codecov uploader. But this was like almost 5 years ago, and Codecov has since rewritten their uploader 3 times. We've additionally made it less flaky by hardcoding a plain-text token a few months ago.

Looking into the coverage reports, something weird is happening. The if-line is marked as covered partially, both in this PR and on `main`:
* https://app.codecov.io/gh/pytest-dev/pytest/pull/12875/blob/testing/code/test_excinfo.py#L966
* https://app.codecov.io/gh/pytest-dev/pytest/blob/main/testing%2Fcode%2Ftest_excinfo.py#L966

However, the `return` line is reported as 100% covered in the PR (https://app.codecov.io/gh/pytest-dev/pytest/pull/12875/blob/testing/code/test_excinfo.py#L971) while on `main` it's marked as not covered (https://app.codecov.io/gh/pytest-dev/pytest/blob/main/testing%2Fcode%2Ftest_excinfo.py#L966).

AFAIU, the `raise` instruction can only be reached on `if True` and the `return` statement would be reachable on `if False`. That should cover both branches of the if-block. So I don't understand why the conditional is marked with partial coverage.

`_makepath()` is also fully covered in both cases:
* https://app.codecov.io/gh/pytest-dev/pytest/pull/12875/blob/src/_pytest/_code/code.py#L955
* https://app.codecov.io/gh/pytest-dev/pytest/blob/main/src%2F_pytest%2F_code%2Fcode.py#L958

P.S. While looking into Codecov, I realized that it does not show any coverage for `src/pytest/` — it's just not there. It looks like we also have Coveragepy misconfigured or something. Something to investigate.

Reply:
I investigated a little, it was introduced in 79ae86cc3f76d69460e1c7beca4ce95e68ab80a6 so that this function did not raise accidentally when inspecting on some env. So this is still useful as it's not only applicable for windows py38.